software/contrib/egressus_melodiam.py

```diff
@@ -0,0 +1,969 @@
+        self.averageDiff = self.average(list)
+        return self.bpmFromMs(self.averageDiff)
+
+
+    def average(self, list):
+        """Pythonic mean average function"""
+        myList = list.copy()
+        return sum(myList) / len(myList)
+
+
+    def initCvPatternBanks(self):
+        """Initialize CV pattern banks"""
+        # Init CV pattern banks, one for each output
+        self.cvPatternBanks = [[], [], [], [], [], []]
+        for n in range(self.maxCvPatterns):
+            self.generateNewRandomCVPattern(self)
+        return self.cvPatternBanks
+
+
+    def generateNewRandomCVPattern(self, new=True, activePatternOnly=False):
+        """Generate new CV pattern for existing bank or create a new bank"""
+        # Note: This function is capable of working with multiple pattern banks
+        #  However, due to current memory limitations only one pattern bank is used
+        try:
+            gc.collect()
+            if new:
+                # new flag provided, create new list
+                if activePatternOnly:
+                    self.cvPatternBanks[self.selectedOutput].append(
+                        self.generateRandomPattern(MAX_STEP_LENGTH, 0, MAX_CV_VOLTAGE)
+                    )
+                else:
+                    for pattern in self.cvPatternBanks:
+                        pattern.append(
+                            self.generateRandomPattern(MAX_STEP_LENGTH, 0, MAX_CV_VOLTAGE)
+                        )
+            else:
+                # Update existing list
+                if activePatternOnly:
+                    self.cvPatternBanks[self.selectedOutput][self.CvPattern] = (
+                        self.generateRandomPattern(MAX_STEP_LENGTH, 0, MAX_CV_VOLTAGE)
+                    )
+                else:
+                    for pattern in self.cvPatternBanks:
+                        pattern[self.CvPattern] = self.generateRandomPattern(
+                            MAX_STEP_LENGTH, 0, MAX_CV_VOLTAGE
+                        )
+            return True
+        except Exception:
+            return False
+
+
+    def generateRandomPattern(self, length, min, max):
+        """Generate a random pattern between min and max of the desired length"""
+        self.t = []
+        for i in range(0, length):
+            self.t.append(round(uniform(min, max), 3))
+        return self.t
+
+
+    def main(self):
+        """Entry point - main loop. See inline comments for more info"""
+        while True:
+            self.updateScreen()
+            self.getK1Value()
+            self.getOutputDivision()
+
+            # Cycle through outputs, process when needed
+            for idx in range(len(cvs)):
+                if (
+                    ticks_diff(ticks_ms(), self.lastSlewVoltageOutputTime[idx])
+                    >= self.msBetweenSamples[idx]
+                    and self.running
+                ):
+
+                    try:

```

Comment:
[question]
What, if any, exceptions are you expecting to catch? It looks like this is just for debug logging, but if there aren't any exceptions being raised you shouldn't need this.

Reply:
removed